lib/web/apiserver.go

```diff
@@ -788,33 +788,40 @@ func (h *Handler) pingWithConnector(w http.ResponseWriter, r *http.Request, p ht
 		ServerVersion: teleport.Version,
 	}
 
+	var as webclient.AuthenticationSettings
+	var authPresent bool
 	if connectorName == constants.Local {
-		as, err := localSettings(cap)
+		as, err = localSettings(cap)
 		if err != nil {
 			return nil, trace.Wrap(err)
 		}
-		response.Auth = as
-		return response, nil
+		authPresent = true
 	}
 
 	// first look for a oidc connector with that name
 	oidcConnector, err := authClient.GetOIDCConnector(r.Context(), connectorName, false)
-	if err == nil {
-		response.Auth = oidcSettings(oidcConnector, cap)
-		return response, nil
+	if !authPresent && err == nil {
+		as = oidcSettings(oidcConnector, cap)
+		authPresent = true
 	}
 
 	// if no oidc connector was found, look for a saml connector
 	samlConnector, err := authClient.GetSAMLConnector(r.Context(), connectorName, false)
-	if err == nil {
-		response.Auth = samlSettings(samlConnector, cap)
-		return response, nil
+	if !authPresent && err == nil {
+		as = samlSettings(samlConnector, cap)
+		authPresent = true
 	}
 
 	// look for github connector
 	githubConnector, err := authClient.GetGithubConnector(r.Context(), connectorName, false)
-	if err == nil {
-		response.Auth = githubSettings(githubConnector, cap)
+	if !authPresent && err == nil {
+		as = githubSettings(githubConnector, cap)
+		authPresent = true
+	}
+
+	if authPresent {
+		as.HasMessageOfTheDay = cap.GetMessageOfTheDay() != ""
+		response.Auth = as
 		return response, nil
 	}
```

Comment:
If you kept the old logic of assigning `response.Auth = xxx` directly above, would you be able to get rid of the extra `authPresent` boolean flag and just use `if response.Auth != nil` check instead? Then you could something like
```suggestion
	if response.Auth != nil {
		response.Auth.HasMessageOfTheDay = cap.GetMessageOfTheDay() != ""
	}
```

which is a bit cleaner.

Reply:
response.Auth isn't nilable at the moment, should I make it a pointer in this pr?